src/NanoMsg.cpp

```diff
@@ -0,0 +1,49 @@
+#include "NanoMsg.h"
+#include <cstring>
+#include <stdexcept>
+#include <g3log/g3log.hpp>
+/**
+ * Constructor for copying a packet hash vector into a nanomsg buffer
+ */
+NanoMsg::NanoMsg(PacketHashVector& data) : sent(false) {
```

Comment:
you are correct, the issue was that the terminology is switched. for nanomsg this is the zero copy api (really one copy). there is a separate api for sending that is available for sending user allocated pointers. i can expose that on the push/pull classes